### PR TITLE
Generate lambda/runtimedef.ml correctly in Swedish

### LINKED ISSUE #10332: Ocaml generat link error if compiling with LC_ALL/LC_COLLATE set to sv_SE.UTF-8 and not C
I have tried to install OCaml with opam(1) and from GitHub source (Any version that opam(1) and OCaml versions like 4.08 and 4.12 and even the latest 4.13 from GitHub).
I have tried on different architectures (x86, AMD64, ARM32) and linux distribution,(Ubuntu, Debina, RaspberryOS, etc) and all fail in the final linking stage for the compiler.

The problem is that in the locale i uses is sv_SE.UTF-8, where "v" and "w" are considered the same character in regular expressions (like s and ss in german or different e in french), so the result somewhere in the compilation stages generate the wrong result.

The solution, which have been tried on all the versions of OCaml, opam(1) and linux distributions can be made to work when telling the computer to use the proper locale C, instead of sv_SE.UTF-8, in regular expressions.  (The proper for compilation, not in other uses)

So to test, first install Swedish locale and try to compile.  Check settings by run locale(1) to print the current LC_* settings.

```
anders@anders-pi:~ $ locale
LANG=sv_SE.UTF-8
LANGUAGE=sv_SE.UTF-8
LC_CTYPE="sv_SE.UTF-8"
LC_NUMERIC="sv_SE.UTF-8"
LC_TIME="sv_SE.UTF-8"
LC_COLLATE="sv_SE.UTF-8"
LC_MONETARY="sv_SE.UTF-8"
LC_MESSAGES="sv_SE.UTF-8"
LC_PAPER="sv_SE.UTF-8"
LC_NAME="sv_SE.UTF-8"
LC_ADDRESS="sv_SE.UTF-8"
LC_TELEPHONE="sv_SE.UTF-8"
LC_MEASUREMENT="sv_SE.UTF-8"
LC_IDENTIFICATION="sv_SE.UTF-8"
LC_ALL=sv_SE.UTF-8
anders@anders-pi:~ $ 

```

Notice LC_COLLATE uses the Swedish locale for things like regular expression and sorting.

Trying to install a switch in opam with this setup will fail with linking errors of missing names.

Adjusting the locale like this will make it work.  First set LC_ALL to nothing, so it doesn't override all other LC_*, then set LC_COLLATE=C.  This will give this output from locale(1)

```
anders@anders-pi:~ $ export LC_ALL=""
anders@anders-pi:~ $ export LC_COLLATE=C
anders@anders-pi:~ $ locale
LANG=sv_SE.UTF-8
LANGUAGE=sv_SE.UTF-8
LC_CTYPE="sv_SE.UTF-8"
LC_NUMERIC="sv_SE.UTF-8"
LC_TIME="sv_SE.UTF-8"
LC_COLLATE=C
LC_MONETARY="sv_SE.UTF-8"
LC_MESSAGES="sv_SE.UTF-8"
LC_PAPER="sv_SE.UTF-8"
LC_NAME="sv_SE.UTF-8"
LC_ADDRESS="sv_SE.UTF-8"
LC_TELEPHONE="sv_SE.UTF-8"
LC_MEASUREMENT="sv_SE.UTF-8"
LC_IDENTIFICATION="sv_SE.UTF-8"
LC_ALL=
anders@anders-pi:~ $ 
```

Now both compiling the latest GitHub source and create any new switch in opam(1) works as it should.

Suggestion to solve this problem:

For compiling OCaml to work on any current or future locale, the proper way is to set LC_ALL="" and LC_COLLATER="C" in the Makefile (or any other better place), so any regular expression in the build phase does what it is supposed to do, explicitly.  Rather to expect the locale choosen by the user do what Ocaml build process want it to do.  It will also still render local adjusted messages etc, as per users wishes.

This error was not easy to find, but by help from some people in the OCaml forum, I manage to get on the right track and find the unexpected error.

### PR DESCRIPTION
**Pull request by dra27**
Given #8985, I'm surprised this hasn't come up already - the problem is very similar.

Fixes #10332.

### CODE REVIEW

**Review by gasche — APPROVED** [review]

**Comment by Octachron** [comment]
(Do we want to fix the manual and documentation scripts too)?

**Comment by dra27** [comment]
Just FTR, the issue manifests itself with glibc 2.28+ (so it's maybe not that surprising that it's only just appearing, given LTS releases)

**Comment by dra27** [comment]
I've tested the fix for the manual and it now builds identically to a non-Swedish build. We don't seem to be testing building the manual any more - @Octachron, is that on purpose or has the actual build of the manual got lost in the Travis->GHA migration?

**Comment by Octachron** [comment]
@dra27 : If you didn't change it on purpose, I think it is fair to say that it was accidentally lost in the GHA transition.

**Comment by dra27** [comment]
@Octachron - in which case, I shall shortly atone for that sin! I think what may have happened is that the new html post-processing PR was going on at the same time, so Travis got switched off with that after the GHA tests were activated!

File: Changes [hunk 1]
```diff
@@ -280,6 +280,10 @@ Working version
   socklen_t detection on Windows.
   (Antonin Décimo, review by David Allsopp and Sébastien Hinderer)
 
+- #10332, #10333: Generate lambda/runtimedef.ml correctly in Swedish locale.
+  (the letter 'w' is not included in '[a-z]' in sv_SE.UTF-8)
+  (David Allsopp, report by Anders Jackson, review by ???)
```
Comment (Octachron) [inline Changes]: The identity of the mysterious `???`  needs to be resolved before merging.

Comment (dra27) [reply Changes]: I have solved for 𝑥!

**Pull request merged**